### ISSUE TITLE
SQLite Performance

### ISSUE THREAD

**Issue by jogibear9988** [sdanyliv/ormbattle#34]
I'm interested in SQLite Performance of the ORMs! Ist it possible that we Include a Test Suite for this?

**Comment by sdanyliv** [comment]
Currently ORM battle uses only SQL Server for tests because many of ORMs supports that database.
I have to investigate if it is possible to use that solution for SQLite testing. 
Originally that project was developed 5 or 6 years ago and i have cloned just for proving speed and maturity of new libraries, especially [linq2db](https://github.com/linq2db/linq2db).

Also which libraries do you propose for test with SQLite?

**Comment by jogibear9988** [comment]
i'd like to see nhibernate, entityframework and linq2db! they all support sqlite!

**Comment by jogibear9988** [comment]
and could you provide current results with mssql? (i dont have mssql installed!)

**Comment by sdanyliv** [comment]
I will provide soon. 
But, i think just, testing ORMs for SQLite will have no benefits because they use same libraries for accessing SQLite. Winner is library that has good Linq support and better materialization speed.

I will run tests for SQL Server today and commit them.

**Comment by sdanyliv** [comment]
First test results are here

https://github.com/sdanyliv/ormbattle/blob/master/_Reports/Output.txt

**Comment by jogibear9988** [comment]
Is it possible to tell wich tests work for bltoolkit and fail for linq2db? Because it's the successor of bltoolkit this should not be!

**Comment by jogibear9988** [comment]
Is it possible to refresh results with current linq2db?

**Comment by sdanyliv** [comment]
Yes, i'll update libraries and run tests again. I think on Monday.

**Comment by jogibear9988** [comment]
I've tested against Northwind.sqlite with current myget

**Comment by jogibear9988** [comment]
<img width="520" alt="bildschirmfoto 2017-02-25 um 23 10 28" src="https://cloud.githubusercontent.com/assets/364896/23335175/d340c5c4-fbaf-11e6-9cbd-8381791ec51a.png">

**Comment by jogibear9988** [comment]
Maybe we can also add "ef core" as a testable framework?

**Comment by jogibear9988** [comment]
To get ormbattle.et working with northwind.sqlite & linq2db, you to change:

in Northwind.cs 

    [InheritanceMapping(Code="True",  Type=typeof(DiscontinuedProduct))]
	[InheritanceMapping(Code="False", Type=typeof(ActiveProduct), IsDefault=true)]

to 

    [InheritanceMapping(Code=true,  Type=typeof(DiscontinuedProduct))]
	[InheritanceMapping(Code=false, Type=typeof(ActiveProduct), IsDefault=true)]

**Comment by jogibear9988** [comment]
Here the Northwind Sqlite Database:

[Northwind.zip](https://github.com/sdanyliv/ormbattle/files/809974/Northwind.zip)

**Comment by jogibear9988** [comment]
Could you rerun the tests with new Linq2Db, so we see if we still have same perf? And also, are sqlite tests possible?

**Comment by sdanyliv** [comment]
Looking for time to do this
